include/api/CAnomalyJobConfig.h

```diff
@@ -169,6 +308,20 @@ class API_EXPORT CAnomalyJobConfig {
                                              core_t::TTime defaultDuration);
 
     private:
+        // Convenience method intended for use by the unit tests only
+        void addDetector(const std::string& functionName,
+                         const std::string& fieldName,
+                         const std::string& byFieldName,
+                         const std::string& overFieldName,
+                         const std::string& partitionFieldName,
+                         const TStrVec& influencers = {},
+                         const std::string& summaryCountFieldName = "") {
```

Comment:
If you add the helper method in the unit test directory to call this then these two arguments won't need to be optional here.